pyro/contrib/timeseries/gp.py

```diff
@@ -253,3 +267,159 @@ def forecast(self, targets, dts):
         filtering_state = self._filter(targets)
         predicted_mean, predicted_covar = self._forecast(dts, filtering_state)
         return MultivariateNormal(predicted_mean, predicted_covar)
+
+
+class DependentMaternGP(TimeSeriesModel):
+    """
+    A time series model in which each output dimension is modeled as a univariate Gaussian Process
+    with a Matern kernel. The different output dimensions become correlated because the Gaussian
+    Processes are driven by a correlated Wiener process; see reference [1] for details.
+    The targets are assumed to be evenly spaced in time. Training and inference are logarithmic
+    in the length of the time series T.
+
+    :param float nu: The order of the Matern kernel; must be 1.5.
+    :param float dt: The time spacing between neighboring observations of the time series.
+    :param int obs_dim: The dimension of the targets at each time step.
+    :param torch.Tensor log_length_scale_init: optional initial values for the kernel length scale
+        given as a `obs_dim`-dimensional tensor
+    :param torch.Tensor log_obs_noise_scale_init: optional initial values for the observation noise scale
+        given as a `obs_dim`-dimensional tensor
+
+    References
+    [1] "Dependent Matern Processes for Multivariate Time Series,"
+        Alexander Vandenberg-Rodes, Babak Shahbaba
+    """
+    def __init__(self, nu=1.5, dt=1.0, obs_dim=1,
+                 log_length_scale_init=None, log_obs_noise_scale_init=None):
+
+        if nu != 1.5:
+            raise NotImplementedError("The only supported value of nu is 1.5")
+
+        self.dt = dt
+        self.obs_dim = obs_dim
+
+        if log_obs_noise_scale_init is None:
+            log_obs_noise_scale_init = -2.0 * torch.ones(obs_dim)
+        assert log_obs_noise_scale_init.shape == (obs_dim,)
+
+        super().__init__()
+
+        self.kernel = MaternKernel(nu=nu, num_gps=obs_dim,
+                                   log_length_scale_init=log_length_scale_init)
+        self.full_state_dim = self.kernel.state_dim * obs_dim
+
+        # we demote self.kernel.log_kernel_scale from being a nn.Parameter
+        # since the relevant scales are now encoded in the wiener noise matrix
+        del self.kernel.log_kernel_scale
+        self.kernel.register_buffer("log_kernel_scale", torch.zeros(obs_dim))
```

Comment:
perhaps there's a better solution here?

Reply:
solution is good, but nit: might as well store `kernel_scale` if it is constant.

Reply:
do you want to fix this now, or should I merge?